src/Stack/Types/Dependency.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Stack.Types.Dependency where
+
+import Stack.Prelude
+import Distribution.Types.VersionRange (VersionRange)
+import Stack.Types.Version (intersectVersionRanges)
+
+
+-- | The value for a map from dependency name. This contains both the
+-- version range and the type of dependency, and provides a semigroup
+-- instance.
+data DepValue = DepValue
+  { dvVersionRange :: !VersionRange
+  , dvType :: !DepType
+  }
+  deriving (Show,Typeable)
```

Comment:
Please could there be a space after the comma? That would be consistent within this module and with other Stack modules. (I appreciate that consistency was missing from the code you moved!)